airflow_dbt/hooks/base.py

```diff
@@ -0,0 +1,164 @@
+        select/models
+    :type exclude: str
+    :param select: Flag used to choose a node or subset of nodes to apply
+        the command to (v0.21.0 and higher)
+    :type select: str
+    :param selector: Config param to reference complex selects defined in
+        the config yaml
+    :type selector: str
+    :param output: {json,name,path,selector}
+    :type output: str
+    :param output_keys: Which keys to output
+    :type output_keys: str
+    :param host: Specify the host to listen on for the rpc server
+    :type host: str
+    :param port: Specify the port number for the rpc server
+    :type port: int
+    :param fail_fast: Stop execution upon a first test failure
+    :type fail_fast: bool
+    :param args:
+    :type args:
+    :param no_compile: Do not run "dbt compile" as part of docs generation
+    :type no_compile: bool
+    """
+    dbt_command_config_annotations = DbtCommandConfig.__annotations__
+    if not dbt_bin or not command:
+        raise ValueError("dbt_bin and command are mandatory")
+    command_params = []
+    for key, value in params.items():
+        # check that the key belongs to DbtCommandConfig keys
+        if key not in dbt_command_config_annotations.keys():
+            raise ValueError(f"{key} is not a valid key")
+        if value is not None:
+            # check that the value has the correct type from dbt_command_config_annotations
+            if type(value) != dbt_command_config_annotations[key]:
+                raise TypeError(f"{key} has to be of type {dbt_command_config_annotations[key]}")
+            # if the param is not bool it must have a non null value
+            cli_param_from_kwarg = "--" + key.replace("_", "-")
+            command_params.append(cli_param_from_kwarg)
+            if type(value) is str:
+                command_params.append(value)
+            elif type(value) is int:
+                command_params.append(str(value))
+            elif type(value) is dict:
+                command_params.append(json.dumps(value))
+            elif type(value) is bool:
+                if not value:
+                    raise ValueError(
+                        f"`{key}` cannot be false. Flags will be passed always "
+                        f"afirmatively. If you want to use a negative flag "
+                        f"such as --no-use-colors then provide "
+                        f"`no_use_colors=True`")
+    return [dbt_bin, command] + command_params
+
+
+class DbtBaseHook(BaseHook, ABC):
+    """
+    Simple wrapper around the dbt CLI and interface to implement dbt hooks
+    """
+
+    def __init__(self, env: Dict = None):
+        """
+        :param env: If set will be passed over to cloud build to run in the
+            dbt step
+        :type env: dict
+        """
+        super().__init__()
+        self.env = env if env is not None else {}
+
+    @abstractmethod
+    def run_dbt(self, dbt_cmd: Union[str, List[str]]):
+        """Run the dbt command"""
```

Comment:
I suppose this works, because the tests pass, but I'm used to seeing a body of a function at least with the place holder `pass` if otherwise empty.

is it not needed?

```suggestion
        """Run the dbt command"""
        pass
```

Reply:
You are definitely right. Let's set it to 'return' as the docs mention